### PR TITLE
Add batch query and attention mask support for GPT2Tokenizer

### PR DESCRIPTION
**Pull request by Toudsour**
GPT2Tokenizer:
-  support batch query
- support attention mask 
- support specifying padding length

Disable part of test due to #74

### CODE REVIEW

File: test/test_gpt2tok.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 from pathlib import Path
 from onnx import helper, onnx_pb as onnx_proto
-from transformers import GPT2Tokenizer
+from transformers import *
```
Comment (wenbingl) [inline test/test_gpt2tok.py:7]: Don't import everything, especially for this large package.

File: tokenizer/gpt2_tokenizer.hpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#include <kernels/ustring.hpp>
```
Comment (wenbingl) [inline tokenizer/gpt2_tokenizer.hpp:1]: What's the idea of extracting this header file?

File: tutorials/tf2onnx_custom_ops_tutorial.ipynb [hunk 3]
```diff
@@ -69,10 +69,10 @@
    "source": [
     "class Model1(tf.keras.Model):\n",
     "\n",
-    "    def __init__(self, name='model1', **kwargs):\n",
```
Comment (wenbingl) [inline tutorials/tf2onnx_custom_ops_tutorial.ipynb:72]: It's better to separate this change into another PR.

File: test/test_gpt2tok.py [hunk 4]
```diff
@@ -23,90 +23,103 @@ def _get_test_data_file(*sub_dirs):
...
+        self._run_tokenizer(["#$%^&()!@?><L:{}\\[];',./`ǠǡǢǣǤǥǦǧǨ"])
+        self._run_tokenizer(["ڠڡڢڣڤڥڦڧڨکڪګڬڭڮگ"])
+        self._run_tokenizer(["⛀⛁⛂⛃⛄⛅⛆⛇⛈⛉⛊⛋⛌⛍⛎⛏"])
+        self._run_tokenizer(["I can feel the magic, can you?", "Yes I do."])
+        self._run_tokenizer(["I can feel the magic, can you?", "Yes I do."], 100)
+
+        enable_custom_op(True)
+
+
+# def test_tokenizer_pyop(self):
```
Comment (wenbingl) [inline test/test_gpt2tok.py:114]: No pyop support any more?

File: test/test_gpt2tok.py (hunk 1 above)
Comment (Toudsour) [reply test/test_gpt2tok.py]: Sure, thanks.

File: tokenizer/gpt2_tokenizer.hpp (hunk 2 above)
Comment (Toudsour) [reply tokenizer/gpt2_tokenizer.hpp:1]: - Keep the same format with other ops
- Let GPT2Tokenizer could be registered by `operator_lists`
I noticed that GPT2Tokenizer is registered by `LoadTokenizerSchemaList`, should we keep this registration way?

File: tutorials/tf2onnx_custom_ops_tutorial.ipynb (hunk 3 above)
Comment (Toudsour) [reply tutorials/tf2onnx_custom_ops_tutorial.ipynb]: Oh, those parts of code are changed by my IDE in a mistake, I will revert it.

File: test/test_gpt2tok.py (hunk 4 above)
Comment (Toudsour) [reply test/test_gpt2tok.py:114]: No, it's just disabled temporarily,  due to #74

File: tokenizer/gpt2_tokenizer.hpp (hunk 2 above)
Comment (wenbingl) [reply tokenizer/gpt2_tokenizer.hpp:1]: > * Keep the same format with other ops
> * Let GPT2Tokenizer could be registered by `operator_lists`
>   I noticed that GPT2Tokenizer is registered by `LoadTokenizerSchemaList`, should we keep this registration way?

The operator list will bring a lot of trouble if we want to use ifdef to enable/disable the operator because of the header file dependency. We should create some functions like LoadTokenizerSchemaList to decouple all operators.

Comment (Toudsour) [reply tokenizer/gpt2_tokenizer.hpp:1]: I think header won't bring trouble because compiler will remove those function/class are not referred by the exported functions if we set right option.

```c++
#include "GPT2Tokenizer.hpp"

class GPT2Tokenizer m_gpt2;

auto register_list = [
#ifdef INCLUDE_GPT2
 &m_gpt2
#endif
]
```
If we turn off `INCLUDE_GPT2`, the whole GPT2Tokenizer will not be included into the final binary, even it's included in a cpp. Compiler will think class GPT2Tokenizer is useless and remove it freely, because there is not any import code(exported function) refer it.

Comment (wenbingl) [reply tokenizer/gpt2_tokenizer.hpp:1]: GCC-liked library is a little complicated than you think. And In the most case, there are a lot of dependency file included in the header file, if there is a third party library, it will introduces a lot of ifdef in the file.

So the load function is the simple way to do it and prevent the user from missing ifdef in the file. And this library, there is no need to let the operator referenced by each other, and no need for the header file at all.

Comment (Toudsour) [reply tokenizer/gpt2_tokenizer.hpp]: I move the content of header into cpp.

**Review by wenbingl — APPROVED** [review]

**Review by wenbingl — APPROVED** [review]
